actions.go

```diff
@@ -2,11 +2,270 @@ package neo4j
 
 import (
 	"fmt"
+	"strings"
 
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
 	"github.com/cloudprivacylabs/opencypher/graph"
 	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
 )
 
+type neo4jAction interface {
+	Queue(neo4j.Transaction, *JobQueue) error
+	Run(neo4j.Transaction, *JobQueue) error
+}
+
+// type neo4jQueue interface {
+// 	Queue(neo4j.Transaction, *JobQueue) error
+// }
+
+type JobQueue struct {
+	queueNodes createNodes
+	queueEdges createEdges
+	actions    []neo4jAction
+	nodeBatch  int64
+	edgeBatch  int64
+}
+
+type createNodes struct {
+	nodes []graph.Node
+}
+
+type createEdges struct {
+	edges []graph.Edge
+}
+
+type DeleteEntity struct {
+	Config
+	graph.Graph
+	entityId int64
+	dbIds    []int64
+}
+
+type CreateEntity struct {
+	Config
+	graph.Graph
+	graph.Node
+	vars map[string]interface{}
+}
+
+func (q *JobQueue) Run(tx neo4j.Transaction) (map[graph.Node]int64, error) {
+	// Go through the jobqueue, create all nodes in batches, get a node-id map
```

Comment:
Run(tx, batchSize int,nodeMap[graph.Node]uint64)

For deleted nodes, delete the id from nodemap
For inserted nodes, add the node with its id to nodemap
The nodemap will be used for linking once everything is done.